components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -339,11 +339,19 @@ RewardsServiceImpl::~RewardsServiceImpl() {
 }
 
 void RewardsServiceImpl::ConnectionClosed() {
+  if (!ready_->is_signaled()) {
+    BLOG(0, "Ledger process disconnected during startup");
+    return;
```

Comment:
In this case, `ledger_factory_` remains bound, so the next time `StartLedgerProcessIfNecessary()` runs, it'll call `CreateLedger()` on a bound, but disconnected `mojo::Remote<mojom::LedgerFactory>`.

Reply:
I think at the very least `ledger_factory_.reset();` is required.

Reply:
Also, looking forward to getting to the changes where it actually makes sense to have a separate `mojom::LedgerFactory` and a `mojom::Ledger`. 🙂

Reply:
I'd be fine with removing this branch and always restarting after a 10 second delay (which is what Ads is doing). Does that sound good?

Reply:
Yeah, sounds good. As for the 10-sec delay, I see how it can make sense to wait for the disconnect messages to arrive asynchronously in their case (they call `reset_on_disconnect()` on their endpoints), but not in ours, as we don't call it, nor do we set a disconnect handler on our endpoints. But anyway, raising the delay shouldn't hurt.

Reply:
Done - I've also added some missing `!Connected` checks that I discovered when stopping the ledger process.